Prepare for version 3

weierophinney opened the pull request:
This patch prepares the package for version 3. Specifically:
- Updates the minimum required PHP version to 5.6.
- Updates to the latest stable release of existing packages.
- Adds new v3-specific components:
  - zend-mvc-console
  - zend-mvc-plugins
  - zend-mvc-form
  - zend-mvc-i18n
  - zend-json-server
  - zend-xml2json
  - zend-diactoros
  - zend-stratigility
  - zend-psr7bridge
  - zend-hydrator
  - zend-servicemanager-di
- Updates the zend-ldap suggestion to specify a known-good stable version of the component.
- Removes the various `bin/*` scripts. The plugin map and class map generators are no longer relevant, and the template map generator is now part of zend-view.

weierophinney committed:
Removed binaries from repository

- plugin map generator and classmap generator are no longer relevant
- template map generator is moving to zend-view

weierophinney committed:
Specify develop branch as 3.0-dev

weierophinney committed:
Raise minimum supported PHP version to 5.6

weierophinney committed:
Updated component list

- Updated to list latest stable version of each
- Added new components:
  - zend-mvc-console
  - zend-mvc-plugins
  - zend-mvc-form
  - zend-mvc-i18n
  - zend-json-server
  - zend-xml2json
  - zend-diactoros
  - zend-stratigility
  - zend-psr7bridge
  - zend-hydrator
  - zend-servicemanager-di
- Updated suggestion list
  - detail stable version of ldap

weierophinney committed:
Updates all documents related to the package

weierophinney committed:
Mark package as an actual Composer metapackage (vs. pseudo)

weierophinney committed:
Update zend-log dep to 2.9

weierophinney committed:
Updated zend-mvc minimum version to 3.0.1

- Brings in the LazyControllerFactory

akrabat commented on CONTRIBUTING.md:
```diff
@@ -5,80 +5,89 @@
-- [zend-view](https://github.com/zendframework/zend-view)
-- [zend-xmlrpc](https://github.com/zendframework/zend-xmlrpc)
-- [ZendXml](https://github.com/zendframework/ZendXml)
+- [zendframework/zend-authentication](https://github.com/zendframework/zendframework/zend-authentication)
```
Too many `zendframework`s here!

weierophinney replied:
Fixed; thanks, @akrabat

akrabat commented on CONTRIBUTING.md:
```diff
@@ -5,80 +5,89 @@
-We will work with you to verify the vulnerability and patch it as soon as possible.
+If you have encountered a potential security vulnerability, please **DO NOT**
+report it on public issue trackers; instead, send it to us at
+[zf-security@zend.com](mailto:zf-security@zend.com) instead.  We will work with
```
Don't need double spaces after a full stop in this day an age :)

weierophinney replied:
Surprisingly, that was `vim`'s auto-formatting!

weierophinney committed:
Fixed component lists

- Removed redundant `zendframework/` path segment in all links
- Removed `zendframework/` prefix from all package names

weierophinney committed:
Replaced full stop + double space with full stop + single space

akrabat commented on README.md:
```diff
@@ -1,26 +1,37 @@
 
 ### INSTALLATION
 
+We no longer recommend installing this package directly. The package is a
```
I'd like to see the recommended installation process noted here - presumably we recommend starting with the skeleton?

weierophinney replied:
Added!

akrabat commented on README.md:
```diff
@@ -30,7 +41,7 @@ $ composer require zendframework/zendframework
 QuickStart:
 
-http://framework.zend.com/manual/current/en/user-guide/overview.html
+- https://docs.zendframework.com/tutorials/getting-started/overview/
```
Why not link the "Zend Framework QuickStart" text above?

weierophinney committed:
Incorporate feedback from @Akrabat

- Link "Zend Framework QuickStart" directly (instead of additional bullet point)
- Provide *recommended* ways to install (zend-mvc and Expressive skeletons)

akrabat commented:
Reviewed. Good to go now. 👍

akrabat commented on composer.json:
```diff
@@ -1,79 +1,85 @@
-    "description": "Zend Framework 2",
-    "type": "library",
+    "description": "Zend Framework",
+    "type": "metapackage",
```
Do we need to update .gitattributes to ensure that we don't install anything or is composer not that strict?

weierophinney replied:
Nope - just confirmed that when you install a metapackage, nothing related to the package is installed in the `vendor/` directory — only the packages listed as requirements are installed. As such, the end-user will have no files from this repository installed at all. :smile:

mwillbanks commented:
LGTM... although, how about a migration guide?

mwillbanks commented on README.md:
```diff
@@ -72,63 +96,69 @@ This package is a metapackage aggregating the following components:
 - [zend-xmlrpc](https://github.com/zendframework/zend-xmlrpc)
-- [ZendXml](https://github.com/zendframework/ZendXml)
+- [zendxml](https://github.com/zendframework/zendxml)
 
```
Add a section on migration...

```markdown
### Migrating from v2 to v3

Zend Framework v2 to v3 has been intended as an incremental upgrade. We have even made efforts in the past year to provide forwards compatibility features in v2 versions of components, to allow users to prepare their code for upgrade.

[View the migration guide](https://docs.zendframework.com/tutorials/migration/to-v3/overview/).
```

weierophinney committed:
Link to the migration guide
